src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -273,42 +275,44 @@ private boolean reinitializeTracer(boolean force) {
         }
 
         try {
-            logger.info("Re/Initialize Datadog-Plugin Tracer: hostname = " + this.hostname + ", traceCollectionPort = " + this.traceCollectionPort);
-            final DDTracer.DDTracerBuilder tracerBuilder = DDTracer.builder();
-            tracerBuilder
-                    .sampler(new ForcePrioritySampler(PrioritySampling.SAMPLER_KEEP))
-                    .writer(DDAgentWriter.builder()
-                    .agentHost(this.hostname)
-                    .traceAgentPort(traceCollectionPort)
-                    .monitoring(Monitoring.DISABLED)
-                    .prioritization(Prioritization.ENSURE_TRACE).build());
-
-            final Tracer ddTracer = tracerBuilder.build();
-            traceBuildLogic = new DatadogTraceBuildLogic(ddTracer);
-            tracePipelineLogic = new DatadogTracePipelineLogic(ddTracer);
+            this.stopAgentHttpClient();
```

Comment:
Move this out of the exception block as you are catching all exceptions inside. As it is now you're calling it again in this block's catch so it looks like a bug in exception handling.

Reply:
Moved outside.

> As it is now you're calling it again in this block's catch so it looks like a bug in exception handling.
Why a bug? I want to be sure that the agent client is stopped if an exception happened.